.github/actions/preparation/action.yml

```diff
@@ -0,0 +1,25 @@
+name: "Preparation"
+description: "Preparation steps to run tests"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout source
+      uses: actions/checkout@v3
+      with:
+        submodules: recursive
+    - name: Install poetry
```

Comment:
I once ran into problems when I installed poetry before python. Because poetry will then be installed using a built-in (probably old) python version, and will keep that version "forever". And if you then install the dependencies, it will fail, because it doesn't take the python version that you want.

Reply:
Currently the setup is:
```
      - name: Checkout source
        uses: actions/checkout@v3
        with:
          submodules: recursive


      - name: Install poetry
        run: | 
          curl -sSL https://install.python-poetry.org | python3 -
          poetry self add poetry-exec-plugin


      - name: Set up Python 3.11
        uses: actions/setup-python@v4
        with:
          python-version: 3.11
          # attempt to restore dependencies from cache (if successful, the following `poetry install` will be a no-op)
          cache: poetry


      - name: Install Python dependencies
        run: poetry install
```

I use this now - with that I should be ok, right?

Reply:
yes, that should be fine.
I don't remember why I thought that python must be installed before poetry. In one repo, it really caused me problems, but apparently, it works here. So never mind my comment from before.

Reply:
Fingers crossed...